### DISCUSSION TITLE
Reformat new Drafts and Logging page to increase usability on landscape monitors

### DISCUSSION THREAD

**Discussion by MagpieFourtyTwo** [2Abendsegler/GClh#2047] [Ideas]
The new Drafts and Logging pages are a pain in the a..., but most obviously we have to cope with them after February, 15th.

And as a) the web designers from gc.com seem to be convinced that the majority of users write their logs on smartphones (why else would they make the decision to design the new pages for portrait mode only?!), and it b) seems to be unknown to their web developer pros that there exist several ways to distinguish between a mobile and a desktop or laptop computer resp. monitor, or c) even more basic between portrait and landscape mode, two thirds of the screen of a "normal" logging working place will simply remain white in the near future ... :/

Thus I badly hope, it will hopefully be possible to "redesign" these pages with the focus on usability not only on smartphones, but on home computers, too.

Therefore: Would it be possible to either ...
- widen the used window size of esp. the "Log this geocache" screen (as currently it's simply impossible to see the preview while writing, once you log more than a few lines - or ...
- display the preview window not *below* but *besides* the compose window ...?
- rearrange the "Drafts" entries more horizontally, i. e. in a way which utilizes the available width, in order to see more list entries at a glance, instead of just putting the data in a narrow list ...

TMALSS: Would be great if these pages will become usable with a landscape monitor, too. ;)

**Comment by MagpieFourtyTwo** [comment]
Moreover I wold prefer to get a one-click-templates list like in the old logging page, instead of having to firt open a drop down ...

But what's you opinion on the mentioned things ...?  ;)

**Comment by 2Abendsegler** [comment]
Moin,

ich muss zugeben, dass ich im Moment nicht wirklich viel Lust und Zeit habe die Seiten komplett umzubauen, auch wenn ich alle deine Punkte gut nachvollziehen kann.

Wir werden uns überlegen, was wir von deinen Punkten auf der neuen Draftseite und gegebenenfalls auch auf der neuen Logseite umsetzen wollen. Aber erst wenn GS mit den Entwicklungen fertig zu sein scheint. 

Außerdem werden wir prüfen, ob es nicht doch einen Weg gibt, von der neuen Draftseite die alte Logseite aufzurufen. Ich glaube das hatten wir vor einer ganzen Weile schon mal erfolglos versucht, aber wir lernen ja auch dazu. Sollte das funktionieren, werden wir die neue Logseite nicht anfassen. Das wäre meines Erachtens die bessere Lösung, weil wir auf der neuen Logseite immer noch das Problem haben, dass beim Einfügen der Logtemplates bzw. der Smilies intern keine Aktualisierung des Logtextes stattfindet, was dazu führt, dass beim Posten diese Änderungen fehlen, sofern nicht noch eine weitere manuell Eingabe erfolgt ist.    

LG

**Comment by MagpieFourtyTwo** [comment]
Nachvollziehbar - vor allem wenn man sich anschaut, wie gc.com das ganze umsetzt ... einfach nur grauenvoll.

Ich persönlich würde einen auch weiterhin bestehenden Zugang zur alten Drafts- und vor allem Log-Seite natürlich auch begrüssen, aber die Hoffnung habe ich in vorige Tage schon aufgegeben, als ich darüber gestolpert bin, dass man in der alten Draft-Liste nicht mehr löschen kann. Augenscheinlich sind da also sogar schon Sachen "ausgebaut" bzw. nicht mehr funktional, so dass ich fürchte, dass man sich früher oder später doch mit den "increased mobile experiences" auseinandersetzen muss.  :/

Das erwähnte Problem mit der "ausbleibenden internen Aktualisierung des Logtextes beim Einfügen von Smileys und/oder Templates" kann ich nicht nachvollziehen, das scheint bei mir zumindest zu funktionieren - hast Du mir da evtl. ein Test-Szenario?

**Reply by 2Abendsegler** [reply]
> Das erwähnte Problem mit der "ausbleibenden internen Aktualisierung des Logtextes beim Einfügen von Smileys und/oder Templates" kann ich nicht nachvollziehen, das scheint bei mir zumindest zu funktionieren - hast Du mir da evtl. ein Test-Szenario?

Verwende beispielsweise das Template [Last cache log], das Template wird eingefügt, mach danach keine Eingaben mehr, dann posten. Im Log sollte jetzt der Inhalt stehen, der vor dem Template vorhanden war. Das Template sollte nicht übernommen worden sein, obwohl es zuvor angezeigt wurde.

**Comment by MagpieFourtyTwo** [comment]
Dann tritt das nur auf, wenn man nach dem Einfügen keine Edit-Aktion mehr durchführt? Ok, das würde erklären, warum es bei mir tut, denn ich editiere nach dem Einfügen immer noch mal.

**Reply by 2Abendsegler** [reply]
> Dann tritt das nur auf, wenn man nach dem Einfügen keine Edit-Aktion mehr durchführt?

Genau. 
Aber das Einfügen kann ja auch in anderer Form passieren. Beispielsweise könntest du noch ein oder mehrere Smilies platzieren, oder du fügst noch einen Standard Satz über das Template Feature ein, weil dir gerade sonst nichts einfällt zu schreiben. Alle Dinge die über den GClh eingefügt werden, ohne dass du danach nochmal manuell editierst, schaffen es nicht ins Log.

**Comment by MagpieFourtyTwo** [comment]
Kommt bei mir nicht vor, ist aber gut zu wissen. Danke!

Solch ein Verhalten kenne ich bisher nur von manchen Passwort-Eingabefeldern, die sich in ähnlicher Weise gegen das AutoFill durch z. B. KeePass sperren (man muss das Passwort dann entweder von Hand eintragen oder nach dem AutoFill mindestens noch mal in das Eingabefeld hinein klicken). Daher war ich davon ausgegangen, dass das eine konfigurierbare Eigenschaft ist ... hab das aber nicht weiter recherchiert. Könnte das sowas sein?

Wenn nicht, erübrigt sich eine Antwort, Du hast sicher auch so genug zu tun, ohne ohne auch noch Stellung zu meinen wilden Vermutungen zu nehmen. ;)

Auf jeden Fall aber möchte ich zumindest noch meinen Dank für die Entwicklung und Pflege von GClh loswerden, ohne den ich vermutlich schon seit längerer Zeit gar nicht mehr loggen würde.

**Comment by 2Abendsegler** [comment]
Die Idee ist nicht verkehrt, dass es sich um eine konfigurierbare Eigenschaft handeln könnte. Zumindest sollten wir uns die Eigenschaften des Input Feldes im Hinblick darauf mal genau ansehen. Auch könnte man versuchen, das Feld nach unseren Wünschen auszutauschen, sofern man trotzdem die Übergabe des Textes beim Posten sicherstellen kann. Das sind zwei neue Ansätze, also gut dass wir drüber gesprochen haben. 😉

Wenn du den GClh etwas streicheln möchtest, dann darfst du oben rechts auf der GitHub Seite gerne ein Sternchen setzen. 🤭

LG und einen schönen Sonntag.

**Comment by 2Abendsegler** [comment]
[Aus dem Geoclub Forum von Sechsfüssler](https://www.geoclub.de/forum/t/gc-little-helper-ii-v0-11.81650/post-1330294):
> Ich fände ein paar neue Features für die hässliche, unpraktische neue Drafts-Seite, zu der man auf GC.com nun keine Alternative mehr hat, toll. Z.B.
>- einen Link zum Cache-Listing
>- Abstände verkleinern
>- ev. das Schreibstift-Icon wegmachen

**Comment by MagpieFourtyTwo** [comment]
> Drafts-Seite, zu der man auf GC.com nun keine Alternative mehr hat,

You always have the choice.  ;)

My current workaround: Copy the uploaded text from the shi... new draft edit window
and paste it to a "normal" log edit window, which is still good old style - at least yet.

Step by step:
- goto [``Drafts``](https://www.geocaching.com/my/fieldnotes.aspx) and open the Draft of your choice for edit
- ``<Ctrl>A`` ``<Ctrl>C`` to copy your uploaded draft text to clipboard
- click cache link at the upper border of the edit window (right below ``Log this geocache``)
- click button ``Log geocache`` => *old*(!) logging window will open
- ``<Ctrl>A`` ``<Ctrl>V`` to paste your draft text from clipboard
- happily edit your draft in good old style  ;)
- finally goto [``Drafts``](https://www.geocaching.com/my/fieldnotes.aspx) again and delete your draft

HTH.
